server/utils_test.go

```diff
@@ -58,3 +58,87 @@ func TestNormalizePath(t *testing.T) {
 		assert.Equal(t, tc.Expected, normalizePath(tc.Full, tc.BaseURL))
 	}
 }
+
+func TestGetLineNumbers(t *testing.T) {
+	tcs := []struct {
+		input      string
+		start, end int
+	}{
+		{
+			input: "L19",
+			start: 16,
+			end:   22,
+		}, {
+			input: "L19-L23",
+			start: 19,
+			end:   23,
+		}, {
+			input: "L23-L19",
+			start: -1,
+			end:   -1,
+		}, {
+			input: "L",
+			start: -1,
+			end:   -1,
+		}, {
+			input: "bad",
+			start: -1,
+			end:   -1,
+		}, {
+			input: "L99-",
+			start: 99,
+			end:   -1,
+		}, {
+			input: "L2",
+			start: 0,
+			end:   5,
+		},
+	}
+	for _, tc := range tcs {
+		start, end := getLineNumbers(tc.input)
+		assert.Equalf(t, tc.start, start, "unexpected start index for getLineNumbers(%q)", tc.input)
+		assert.Equalf(t, tc.end, end, "unexpected end index for getLineNumbers(%q)", tc.input)
+	}
+}
+
+func TestInsideLink(t *testing.T) {
+	tcs := []struct {
+		input    string
+		index    int
+		expected bool
+	}{
+		{
+			input:    "[text](link)",
+			index:    7,
+			expected: true,
+		}, {
+			input:    "[text]( link space)",
+			index:    8,
+			expected: true,
+		}, {
+			input:    "text](link",
+			index:    6,
+			expected: true,
+		}, {
+			input:    "text] (link)",
+			index:    7,
+			expected: false,
+		}, {
+			input:    "text](link)",
+			index:    6,
+			expected: true,
+		}, {
+			input:    "link",
+			index:    0,
+			expected: false,
+		}, {
+			input:    " (link)",
+			index:    2,
+			expected: false,
+		},
+	}
+
+	for _, tc := range tcs {
+		assert.Equalf(t, tc.expected, isInsideLink(tc.input, tc.index), "unexpected result for isInsideLink(%q, %d)", tc.input, tc.index)
```

Comment:
Why are you using `%q` for one variable and `%d` for another?

Reply:
Because `tc.input` is a string and `tc.index` is a int.

Reply:
To double quote string we use %q. See the example.
**Code**
![image](https://user-images.githubusercontent.com/85667960/171417072-fa91c299-fa38-40b4-aed7-0c028a968577.png)
**Output**
![image](https://user-images.githubusercontent.com/85667960/171417155-7bc821be-2e19-4c64-bb63-b7687b1ca5b4.png)